### PR TITLE
Add documentation for Ethereum and Solana feature policies

### PR DESCRIPTION
**Pull request by bbondy**
This PR also provides information about secure contexts.

Credit goes to @diracdeltas and @thypon for most of the content writing from this PR.

It covers functionality added by these pull requests:
- https://github.com/brave/brave-core/pull/13783
- https://github.com/brave/brave-core/pull/13739

### CODE REVIEW

File: docs/ethereum/overview.md [hunk 1]
```diff
@@ -5,7 +5,10 @@ slug: /ethereum
 
 # Overview
 
-Brave injects a `windows.ethereum` provider object on all pages.  
+Brave injects a `windows.ethereum` provider object on all secure sites.
+Secure 1st party iframes will also have `window.ethereum` exposed but will work the same as other secure feature policies.
+For more information please see [this work](https://github.com/brave/brave-browser/issues/23710).
+
```
Comment (darkdh) [inline docs/ethereum/overview.md:11]: This also needs to be added into Solana overview section. Or we can just move it to `provider-objects.md`

**Review by diracdeltas — APPROVED** [review]
lgtm, will ask kyle/joe to look as well

File: docs/provider-availability.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+1. If the iframe is third party to the top-level origin, it will be blocked UNLESS the iframe has the `allow="{solana/ethereum}"` attribute (where “solana" and “ethereum" values control the corresponding API permissions). 
+2. If the iframe is first party to the top-level origin AND the `sandbox` attribute is set on the iframe, it will be blocked UNLESS `sandbox="allow-same-origin"` is set. Note "allow-same-origin"` does nothing if the iframe is third-party.
+3. For security-conscious users, we add a setting to block window.{ethereum,solana} in ALL iframes, regardless of origin or attributes. This matches the default behavior on iOS.
+
+### iOS
+Currently on iOS, `window.ethereum` and `window.braveSolana` are both undefined in all iframes.
+
+## Example cases
+
+In all these cases, the `window.ethereum` or `window.braveSolana` request is coming from the innermost iframe.
```
Comment (kdenhartog) [inline docs/provider-availability.md:38]: Thanks for documenting this list down in here!

File: docs/default-wallet.md [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-sidebar_position: 3
+sidebar_position: 5
 ---
 
 # Default Wallet setting
 
-Dapps work by communicating with a special object that Brave Wallet and extensions like MetaMask provide named `window.ethereum`.
-Only one wallet can provide `window.ethereum` to websites.
-In Brave, we expose a setting in `brave://settings/wallet` to be able to change which wallet provides `window.ethereum`.
+We expose a setting in `brave://settings/wallet` to control how Brave makes the provider objects available (`window.ethereum` and `window.braveSolana` and its alias `window.solana`)
```
[the remark this answers is not in the document]
Comment (bbondy) [reply docs/default-wallet.md:7]: This is somewhat recent but not related to security. To get in Solana's wallet adapter repo they would not accept `window.solana` so we needed to rename to `window.braveSolana` and we retain an alias to that of `window.solana` for now.

https://github.com/solana-labs/wallet-adapter/pull/445#discussion_r904203433

[the remark this answers is not in the document]
Comment (kdenhartog) [reply docs/default-wallet.md:7]: Thanks for linking to that thread, it's not fully clear to me about their motivations but it's excellent context for me if the conversation comes up again in the future.

**Review by kdenhartog — APPROVED** [review]
LGTM!

File: docs/default-wallet.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (bbondy) [reply docs/default-wallet.md:7]: Yep! Apologies for not highlighting that to you earlier.

**Pull request merged**